src/main/java/de/zalando/zmon/dataservice/data/RedisDataPointsQueryStore.java

```diff
@@ -39,16 +61,44 @@ public int store(String query) {
 
     @VisibleForTesting
     byte[] compress(String str) throws IOException {
+
         final byte[] dataToCompress = str.getBytes();
+        byte[] tracePayload = buildRedisTracePayload();
         final ByteArrayOutputStream byteStream = new ByteArrayOutputStream(dataToCompress.length);
+        ByteArrayOutputStream payLoad = new ByteArrayOutputStream(tracePayload.length + dataToCompress.length);
+
         try {
             try (GZIPOutputStream zipStream = new GZIPOutputStream(byteStream, true)) {
                 zipStream.write(dataToCompress);
+                payLoad.write(tracePayload);
+                payLoad.write(byteStream.toByteArray());
+                LOG.debug(byteStream.toString());
+                LOG.debug(payLoad.toString());
             }
         } finally {
             byteStream.close();
+            payLoad.close();
         }
+        return payLoad.toByteArray();
+    }
+
+    byte[] buildRedisTracePayload(){
+
+        SpanContext spanContext = tracer.activeSpan().context();
+        Carrier carrier = new Carrier(new HashMap<>());
+        tracer.inject(spanContext, Format.Builtin.TEXT_MAP, carrier);
+
+        final byte[] firstByte = ByteBuffer.allocate(1).put("0".getBytes()).array();
+        final byte[] context = carrier.toString().getBytes();
```

Comment:
We need to pick a serialization/deserialization approach here - Java default toString won't work as one here.